airbytehq/airbyte
# :bug: low-code: Fix injection of page token in first request

## What
Fix a few issues with the pagination's `inject_on_first_request`. When True, the pagination is expected to inject the page_number of offset in the first request sent to the API.

The issues were:
- [The fields were not properly wired from the model to the python classes](https://github.com/airbytehq/airbyte/pull/32666/files#diff-bbfbb380b441567d8e40967c149ae0397e41357a213f05add6d3b72cf92a91fdR839-R853)
- [The paginator's initial value wasn't properly reset ](https://github.com/airbytehq/airbyte/pull/32666/files#diff-b36b8735005712a942bee18c2b6b9b824d9b08be821b16c4ea527d07e2f8901cR156)

Other changes are fixing mypy issues and tests

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py`
2. `airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/default_paginator.py`
3. `airbyte-cdk/python/unit_tests/sources/declarative/parsers/test_model_to_component_factory.py`
4. `airbyte-cdk/python/unit_tests/sources/declarative/requesters/paginators/test_default_paginator.py`
5. `airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/request_options/request_options_provider.py`

## Testing
To test this change:
1. Run `python main.py read  --config secrets/config.json --catalog integration_tests/configured_catalog.json --debug` with source-aha and verify the first request does not include the page number (`https://airbyte1.aha.io/api/v1/products?per_page=5`)
2. Add `inject_on_first_request: true` to the source-aha's pagination strategy
3. `python main.py read  --config secrets/config.json --catalog integration_tests/configured_catalog.json --debug`
4. Confirm the first request sent includes the page number (`https://airbyte1.aha.io/api/v1/products?page=0&per_page=5`)
5. Set `inject_on_first_request` to false
6. Verify the page number is not included in the first request
